cdn_lambda/functions/origin_response/origin_response.py

```diff
@@ -0,0 +1,63 @@
+import json
+import logging
+import re
+
+LOG = logging.getLogger("origin-response")
+
+
+class LambdaClient(object):
+    def __init__(self, conf_file="lambda_conf.json"):
+        self._conf_file = conf_file
+        self._conf = None
+
+    @property
+    def conf(self):
+        if not self._conf:
+            with open(self._conf_file, "r") as json_file:
+                self._conf = json.load(json_file)
+
+        return self._conf
+
+    def handler(self, event, context):
+        # pylint: disable=unused-argument
+
+        request = event["Records"][0]["cf"]["request"]
+        response = event["Records"][0]["cf"]["response"]
+
+        max_age = self.conf["headers"]["max_age"]
+        max_age_pattern_whitelist = [
+            ".+/PULP_MANIFEST",
+            ".+/listing",
+            ".+/repodata/repomd.xml",
+            ".+/ostree/repo/refs/heads/.*/.*",
+        ]
+
+        try:
+            original_uri = request["headers"]["exodus-original-uri"][0][
+                "value"
+            ]
+
+        except (KeyError):
+            LOG.debug(
+                "Could not add Cache-Control header 'exodus-original-uri'"
+            )
+            original_uri = None
+
+        if original_uri is not None:
+            for pattern in max_age_pattern_whitelist:
+                if re.match(pattern, original_uri):
+                    response["headers"]["cache-control"] = [
+                        {
+                            "key": "Cache-Control",
+                            "value": "max-age={}".format(max_age),
+                        }
+                    ]
```

Comment:
Okay, maybe I'm missing something about structure for headers. Are the "key" and "value" keys necessary?